web/src/engine/websites/ZenManga.ts

```diff
@@ -0,0 +1,93 @@
+﻿import { Tags } from '../Tags';
+import icon from './ZenManga.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON } from '../platform/FetchProvider';
+import * as devalue from 'devalue';
+
+type APIManga = {
+    id: string;
+    slug: string;
+    name: {
+        ru: string;
+    }
+};
+
+type HydratedChapter = {
+    id: string;
+    name: string;
+    number: number;
+    branchId: string;
+    volume: number;
+};
+
+type HydratedBranch = {
+    id: string;
+    publishers: {
+        name: string;
+    }[]
+}
+
+type APIPages = {
+    pages: {
+        image: string;
+    }[]
+};
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://api.zenmanga.io/v2/';
+
+    public constructor() {
+        super('zenmanga', 'ZenManga', 'https://zenmanga.io', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Russian, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/content/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const { slug, name: { ru } } = await FetchJSON<APIManga>(new Request(new URL(`./books/${url.split('/').at(-1)}`, this.apiUrl)));
+        return new Manga(this, provider, slug, ru);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let page = 0, run = true; run; page++) {
+            const mangas = await this.GetMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const data = await FetchJSON<APIManga[]>(new Request(new URL(`./books?page=${page}&size=200&sort=viewsCount%2Cdesc`, this.apiUrl)));
+        return data.map(({ slug, name: { ru } }) => new Manga(this, provider, slug, ru));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const revivers = {
```

Comment:
Declaring resolvers will no longer work as soon as the website introduces custom types (except if we are going to re-implement all the websites custom types)